test/test_package_loading.py

```diff
@@ -105,6 +106,22 @@ def test_load_packages_sequential(selenium_standalone, packages):
     assert selenium.logs.count(f'Loading {packages[1]}') == 1
 
 
+def test_different_ABI(selenium_standalone):
+    original_file = open('build/numpy.js', 'r+')
```

Comment:
Rather than modifying the file, we should copy it and modify it there, and then use the URL method to load the package.   As it is now, this test will fail if we ever were to run tests in parallel (because it temporarily breaks the numpy package for any other tests that might need to use the numpy package).

Reply:
What exactly is meant by the ``URL`` method? Also, let's say I create a modified ``numpy.js`` file in the temporary directory using ``f = tmpdir.join("numpy.js")``. What ``path`` or ``URL`` should I exactly pass in the ``load_package`` function? This is the only bottleneck I am facing. Rest all is done. Please let me know, thanks!

Reply:
Specifically, Let's say the modified numpy.js [with wrong ABI] is created at ``/tmp/pytest-of-root/pytest-9/test_different_ABI_chrome_0/numpy.js``. Do I pass this path in ``load_package``? Doing so does not work as ``load_package`` only accepts URLs ending in ``.js`` OR simply the names such as ``numpy``. How to proceed in this case?

Reply:
I don't think you can copy it to a directory, because the webserver used in testing won't find it there.  It will need to be created in the same directory as the original `numpy.js` (the `build` directory).

It you copy it to a file called `numpy-broken.js`, you should be able to pass `/numpy-broken.js` to `loadPackage`.  (The `/` should trigger the use of the URL, rather than package name, method in `loadPackage`).

Reply:
I did try this before, but then, it results in a ``Invalid package name or URI`` error. This is because ``./numpy-broken.js`` OR ``/numpy-broken.js`` will not satisfy the regex which is used in the ``load_package`` function. Specifically, the section where the regexes are defined aka 
```
var package_name_regexp = '[a-z0-9_][a-z0-9_\-]*'
var package_uri_regexp =
    new RegExp('^https?://.*?(' + package_name_regexp + ').js$', 'i');
var package_name_regexp = new RegExp('^' + package_name_regexp + '$', 'i');
```

As seen here, both of the regexes fail for the input ``./numpy-broken.js`` OR ``/numpy-broken.js``. The URLs are expected to have ``http`` or ``https`` in the beginning. In fact, I have also tried it by running the server locally and trying ``./numpy.js`` OR ``/numpy.js`` which although fulfils the promise, but returns ``undefined`` as seen below:

<img width="873" alt="screenshot 2019-01-08 at 8 35 29 pm" src="https://user-images.githubusercontent.com/20173739/50838838-02cbce80-1385-11e9-9a96-487c68c9a1ce.png">

What should be done? Should we modify the regexes? I also tried passing just ``numpy-broken`` since this file was made inside the ``build`` directory but then it results in a ``Unknown package 'numpy-broken'`` error probably because its not in the ``packages.json`` file.

Reply:
Ah, I forgot the regex is so strict.  You could build up a full http url from `selenium_standalone.host` and `selenium_standalone.port`, though, e.g.:

```
f"http://{selenium_standalone.host}:{selenium_standalone.port}/numpy-broken.js"
```

Reply:
So I tried this out too. It turns out that even this doesn't work unless we have an actual entry for ``numpy-broken.js`` in the ``packages.json`` file. Once we do add an entry for ``numpy-broken.js`` in ``packages.json``, it works. Now, modifying the ``packages.json`` file in the test itself will lead to the same issue - aka - what if when we run the tests in parallel. What should be done in this case?

Reply:
The URL generated is correct and this issue is not related to it. The reason that it doesn't work without adding it to ``packages.json`` file is because the function ``load_package`` is defined that way. It checks if the package we are trying to load is present in this file, else, it returns the error of ``Unknown package 'numpy-broken'``

Reply:
This is the section in the ``while`` loop of the ``load_package`` function that does this check!
````
else {
        console.log(`Loading ${package} from ${package_uri}`);

        toLoad[package] = package_uri;
        if (packages.hasOwnProperty(package)) {
          packages[package].forEach((subpackage) => {
            if (!(subpackage in loadedPackages) && !(subpackage in toLoad)) {
              queue.push(subpackage);
            }
          });
        } else {
          console.error(`Unknown package '${package}'`);
          return;
        }
      }
````

Reply:
Sorry, that's my bad.  I thought if you provided a URL, it wouldn't try to do the dependency resolution using `packages.json`.  @rth: Is that how that's supposed to work?  How does one load a completely unknown package?